pet-store-react-hooks/src/DeletePetModal/DeleteModal.test.tsx

```diff
@@ -0,0 +1,203 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent from '@testing-library/user-event';
+import { HttpResponse, http } from 'msw';
+import { setupServer } from 'msw/node';
+
+import { DeletePetModal } from '~/DeletePetModal/DeletePetModal';
+import { apiBaseUrl } from '~/infrastructure/api-client';
+import { WaitHandle } from '~testing/WaitHandle';
+import { petKindsMap, pets, petsList } from '~testing/data';
+import { handlers } from '~testing/handlers';
+
+const server = setupServer(...handlers);
+
+beforeAll(() => server.listen());
+afterEach(() => server.resetHandlers());
+afterAll(() => server.close());
+
+// Question: When i have pet for testing, is it OK to get it outside of the tests and use it in all of them?
+const petForDelete = petsList.find((x) => x.petId === 44);
+if (!petForDelete) {
+  throw new Error('Pet with petId: 44 is not found in the testing data');
+}
+
+// Question: I'm defining the callback function here and cleaning the mock data in the tests i'm using them. OK?
+const onClose = jest.fn();
+const onDeleted = jest.fn();
+jest.mock('~/infrastructure/reportError');
+
+test('Header and pet information is displayed correctly, Delete and Cancel buttons are visualised', async () => {
+  render(
+    <DeletePetModal
+      pet={petForDelete}
+      petKind={petKindsMap.get(petForDelete.kind)}
+      onClose={onClose}
+      onDeleted={onDeleted}
+    />
+  );
+
+  // Header
+  const headerText = await screen.findByText('Delete pet #44');
+  expect(headerText).toBeInTheDocument();
+
+  const headerCloseButton = await screen.findByRole('button', { name: 'X' });
+  expect(headerCloseButton).toBeInTheDocument();
+
+  // Pet info
+  const petName = await screen.findByText('Name: Kenny');
+  expect(petName).toBeInTheDocument();
+
+  const petKind = await screen.findByText('Kind: Parrot');
+  expect(petKind).toBeInTheDocument();
+
+  const petDateAdded = await screen.findByText('Date added: 27 Oct 2022');
+  expect(petDateAdded).toBeInTheDocument();
+
+  // Buttons
+  const deleteButton = await screen.findByRole('button', { name: 'Delete' });
+  expect(deleteButton).toBeInTheDocument();
+
+  const cancelButton = await screen.findByRole('button', { name: 'Cancel' });
+  expect(cancelButton).toBeInTheDocument();
+});
+
+test('Modal is closed on header close button click', async () => {
```

Comment:
Maybe say `onClose is called` instead of `Modal is closed` since we are not closing the modal but calling a prop instead.